modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -863,6 +867,10 @@ namespace cv {
 
                     std::string activation = getParam<std::string>(layer_params, "activation", "linear");
                     if (activation == "leaky")
+                    {
+                        setParams.setActivation("leaky");
+                    }
+                    else if (activation == "relu")
                     {
                         setParams.setActivation("relu");
```

Comment:
Please replace to 
```cpp
if (activation != "linear")
    setParams.setActivation(activation);
```
because there is already `"Unsupported activation" check inside `setActivation`.